src/metabase/api/card.clj

```diff
@@ -125,6 +128,20 @@
   [_filter-option model-id]
   (cards-for-segment-or-metric :segment model-id))
 
+(defmethod cards-for-filter-option* :stale
+  [_]
+  (let [card-id->field-usages (group-by :card_id (t2/select :model/FieldUsage
+                                                            {:select [:fu.* [:f.name :new_column_name] [:t.name :new_table_name]]
+                                                             :from   [[(t2/table-name :model/FieldUsage) :fu]]
+                                                             :join   [[(t2/table-name :model/Field) :f] [:= :f.id :fu.field_id]
+                                                                      [(t2/table-name :model/Table) :t] [:= :t.id :f.table_id]]
+                                                             :where  [:= false :fu.is_current]}))
```

Comment:
Queries like this are exactly in the gray area of "should this be in the model or the API?" for me. It's like the old Rails debates: fat model or fat controller?

Reply:
> Should this be in the model or the API? It's like the old Rails debates...

I was always more of a Domain Context Interaction guy 😜 - a third way, for the third estate.

Reply:
This does feel like Model code though.